lib/Service/TextBlockService.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Service;
+
+use Html2Text\Html2Text;
+use OCA\Mail\Db\TextBlock;
+use OCA\Mail\Db\TextBlockMapper;
+use OCA\Mail\Db\TextBlockShare;
+use OCA\Mail\Db\TextBlockShareMapper;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\Files\NotPermittedException;
+use OCP\IGroupManager;
+use OCP\IUserManager;
+
+class TextBlockService {
+
+	private TextBlockMapper $textBlockMapper;
+	private TextBlockShareMapper $textBlockShareMapper;
+	private IUserManager $userManager;
+	private IGroupManager $groupManager;
+
+	public function __construct(TextBlockMapper $textBlockMapper,
+		TextBlockShareMapper $textBlockShareMapper,
+		IGroupManager $groupManager,
+		IUserManager $userManager) {
+		$this->textBlockMapper = $textBlockMapper;
+		$this->textBlockShareMapper = $textBlockShareMapper;
+		$this->userManager = $userManager;
+		$this->groupManager = $groupManager;
+	}
+
+	/**
+	 * @param string $userId
+	 * @return TextBlock[]
+	 */
+	public function findAll(string $userId): array {
+		return $this->textBlockMapper->findAll($userId);
+	}
+
+	/**
+	 * @param string $userId
+	 * @return TextBlock[]
+	 * @throws DoesNotExistException
+	 */
+	public function findAllSharedWithMe(string $userId): array {
+		$user = $this->userManager->get($userId);
+		if ($user === null) {
+			throw new DoesNotExistException('User does not exist');
```

Comment:
Let's use a different exception as DoesNotExistException belongs to our database abstraction. Maybe https://www.php.net/manual/en/class.invalidargumentexception.php.

Reply:
It seems we are doing that in many more places in mail, and therefore I'm okay with keeping it. I would still prefer to use other exceptions (like a newly created UserNotFoundException) because reusing exceptions from a different layer still seems weird ;)